### ISSUE TITLE
Panic while running Artifact.Windows.KapeFiles.Targets on 0.6.7+ 

### ISSUE THREAD

**Issue by clayscode** [Velocidex/velociraptor#2956]
Having an issue with all Velociraptor builds 0.6.7 and above where they panic while calling `Artifact.Windows.KapeFiles.Targets`

Specifically, this call fails: 
```sql
Artifact.Windows.KapeFiles.Targets(Device=KDevice, VSSAnalysis="Y", _BasicCollection = "Y", WebBrowsers="Y", Antivirus="Y", RDPCache="Y", WindowsTimeline="Y", UseAutoAccessor="Y")
```

Here is the stacktrace:
```go
{"level":"info","msg":"Starting collection of Custom.Kape_BasicCollection\n","time":"2023-09-20T10:20:52+02:00"}
{"level":"info","msg":"Compiled all artifacts.","time":"2023-09-20T10:20:52+02:00"}
{"level":"info","msg":"Selecting _BasicCollection\n","time":"2023-09-20T10:20:52+02:00"}
{"level":"info","msg":"Selecting Antivirus\n","time":"2023-09-20T10:20:52+02:00"}
{"level":"info","msg":"Selecting RDPCache\n","time":"2023-09-20T10:20:52+02:00"}
{"level":"info","msg":"Selecting WebBrowsers\n","time":"2023-09-20T10:20:52+02:00"}
{"level":"info","msg":"Selecting WindowsTimeline\n","time":"2023-09-20T10:20:52+02:00"}
{"level":"info","msg":"ntfs: Selecting glob $Boot\n","time":"2023-09-20T10:20:52+02:00"}
{"level":"info","msg":"ntfs: Selecting glob $Extend/$UsnJrnl:$J\n","time":"2023-09-20T10:20:52+02:00"}
[...]
[INFO] 2023-09-20T10:21:07+02:00 Collecting file \\.\C:\127133-128-4 into /uploads/mft/%5C%5C.%5CC%3A/Windows/prefetch/UPDATE.EXE-E1E21195.pf (61538 bytes)
panic: runtime error: index out of range [280] with length 280

goroutine 28003 [running]:
www.velocidex.com/golang/go-ntfs/parser.(*NTFS_ATTRIBUTE).RunList(0xc0022d9c20)
        /home/runner/go/pkg/mod/www.velocidex.com/golang/go-ntfs@v0.1.2-0.20221117122413-b97c856cb140/parser/attribute.go:164 +0x2fd
www.velocidex.com/golang/go-ntfs/parser.joinAllVCNs(0xc0010cd380, {0xc001d5b0c0, 0x3, 0x2960000?})
        /home/runner/go/pkg/mod/www.velocidex.com/golang/go-ntfs@v0.1.2-0.20221117122413-b97c856cb140/parser/easy.go:381 +0x157
www.velocidex.com/golang/go-ntfs/parser.OpenStream(0xc0010cd380?, 0xc00151d300?, 0xc003108840?, 0x5e6b?)
        /home/runner/go/pkg/mod/www.velocidex.com/golang/go-ntfs@v0.1.2-0.20221117122413-b97c856cb140/parser/easy.go:349 +0x207
www.velocidex.com/golang/velociraptor/accessors/ntfs.(*MFTFileSystemAccessor).OpenWithOSPath(0xc002511cb0, 0xc003108840)
        /home/runner/work/velociraptor/velociraptor/accessors/ntfs/mft.go:139 +0x12d
www.velocidex.com/golang/velociraptor/vql/networking.(*UploadFunction).Call(0xc002ae0860?, {0x3256360, 0xc002bed4c0}, {0x32701d0, 0xc003073170}, 0x0?)
        /home/runner/work/velociraptor/velociraptor/vql/networking/upload.go:84 +0x36f
www.velocidex.com/golang/vfilter.(*_SymbolRef).callFunction(0xc000c8ecd0, {0x3256360?, 0xc002bed4c0}, {0x32701d0?, 0xc003073170}, {0x324c7d8?, 0x4057b88})
        /home/runner/go/pkg/mod/www.velocidex.com/golang/vfilter@v0.0.0-20221111065201-759cb2572787/vfilter.go:1964 +0x57e
www.velocidex.com/golang/vfilter.(*_SymbolRef).Reduce(0xc000c8ecd0, {0x3256360, 0xc002bed4c0}, {0x32701d0, 0xc003073170})
        /home/runner/go/pkg/mod/www.velocidex.com/golang/vfilter@v0.0.0-20221111065201-759cb2572787/vfilter.go:1818 +0x1c6
www.velocidex.com/golang/vfilter.(*_Value).Reduce(0xc0013a34a0, {0x3256360, 0xc002bed4c0}, {0x32701d0, 0xc003073170})
        /home/runner/go/pkg/mod/www.velocidex.com/golang/vfilter@v0.0.0-20221111065201-759cb2572787/vfilter.go:1637 +0x14f
www.velocidex.com/golang/vfilter.(*_MemberExpression).Reduce(0xc001e2a4c0, {0x3256360, 0xc002bed4c0}, {0x32701d0?, 0xc003073170?})
        /home/runner/go/pkg/mod/www.velocidex.com/golang/vfilter@v0.0.0-20221111065201-759cb2572787/vfilter.go:1231 +0x55
www.velocidex.com/golang/vfilter.(*_MultiplicationExpression).Reduce(0xc001e2a500, {0x3256360, 0xc002bed4c0}, {0x32701d0?, 0xc003073170?})
        /home/runner/go/pkg/mod/www.velocidex.com/golang/vfilter@v0.0.0-20221111065201-759cb2572787/vfilter.go:1564 +0x52
www.velocidex.com/golang/vfilter.(*_AdditionExpression).Reduce(0xc001e2a540, {0x3256360, 0xc002bed4c0}, {0x32701d0?, 0xc003073170?})
        /home/runner/go/pkg/mod/www.velocidex.com/golang/vfilter@v0.0.0-20221111065201-759cb2572787/vfilter.go:1455 +0x52
www.velocidex.com/golang/vfilter.(*_ConditionOperand).Reduce(0xc003290138, {0x3256360, 0xc002bed4c0}, {0x32701d0?, 0xc003073170?})
        /home/runner/go/pkg/mod/www.velocidex.com/golang/vfilter@v0.0.0-20221111065201-759cb2572787/vfilter.go:1502 +0x85
www.velocidex.com/golang/vfilter.(*_OrExpression).Reduce(0xc001e2a580, {0x3256360, 0xc002bed4c0}, {0x32701d0?, 0xc003073170?})
        /home/runner/go/pkg/mod/www.velocidex.com/golang/vfilter@v0.0.0-20221111065201-759cb2572787/vfilter.go:1407 +0x55
www.velocidex.com/golang/vfilter.(*_AndExpression).Reduce(0xc001e2a5c0, {0x3256360, 0xc002bed4c0}, {0x32701d0?, 0xc003073170?})
        /home/runner/go/pkg/mod/www.velocidex.com/golang/vfilter@v0.0.0-20221111065201-759cb2572787/vfilter.go:1365 +0x49
www.velocidex.com/golang/vfilter.(*_AliasedExpression).Reduce(0xc002ae082c?, {0x3256360?, 0xc002bed4c0?}, {0x32701d0?, 0xc003073170?})
        /home/runner/go/pkg/mod/www.velocidex.com/golang/vfilter@v0.0.0-20221111065201-759cb2572787/vfilter.go:732 +0x9e
www.velocidex.com/golang/vfilter.(*_SelectExpression).Transform.func2({0x3256360, 0xc002bed4c0}, {0xc002ae0870?, 0x6?})
        /home/runner/go/pkg/mod/www.velocidex.com/golang/vfilter@v0.0.0-20221111065201-759cb2572787/vfilter.go:986 +0x46
www.velocidex.com/golang/vfilter.MaterializedLazyRow({0x3256360, 0xc002bed4c0}, {0x2a925a0?, 0xc0002dde30?}, {0x32701d0, 0xc003072e10})
        /home/runner/go/pkg/mod/www.velocidex.com/golang/vfilter@v0.0.0-20221111065201-759cb2572787/lazy.go:96 +0x1d0
www.velocidex.com/golang/vfilter.(*_Select).processSingleRow(0xc002ab3480, {0x3256360, 0xc002bed4c0}, {0x32701d0, 0xc0025c7560}, {0x2cfe400, 0xc0025c7560}, 0xc003565bc0)
        /home/runner/go/pkg/mod/www.velocidex.com/golang/vfilter@v0.0.0-20221111065201-759cb2572787/vfilter.go:624 +0x406
www.velocidex.com/golang/vfilter.(*_Select).Eval.func3()
        /home/runner/go/pkg/mod/www.velocidex.com/golang/vfilter@v0.0.0-20221111065201-759cb2572787/vfilter.go:603 +0xe5
created by www.velocidex.com/golang/vfilter.(*_Select).Eval
        /home/runner/go/pkg/mod/www.velocidex.com/golang/vfilter@v0.0.0-20221111065201-759cb2572787/vfilter.go:588 +0x2ca
```

This is on `Windows 11 - 10.0.22621 Build 22621`. On other operating systems (Windows Server 2012, 2016, 2022), we can't replicate the crash. This happens randomly - not limited to prefetch files.

**Comment by scudette** [comment]
This looks like a very old version of the NTFS parser. Can you please confirm this bug also occurs with the current release?

**Comment by clayscode** [comment]
This is with 0.7.0
```go
panic: runtime error: index out of range [304] with length 304

goroutine 37944 [running]:
www.velocidex.com/golang/go-ntfs/parser.(*NTFS_ATTRIBUTE).RunList(0xc001dd3c80)
        /go/pkg/mod/www.velocidex.com/golang/go-ntfs@v0.1.2-0.20230508135129-d367f954db15/parser/attribute.go:158 +0x313www.velocidex.com/golang/go-ntfs/parser.joinAllVCNs(0xc000961dd0, {0xc001efdb00?, 0x12, 0x20})
        /go/pkg/mod/www.velocidex.com/golang/go-ntfs@v0.1.2-0.20230508135129-d367f954db15/parser/easy.go:393 +0x1d3
www.velocidex.com/golang/go-ntfs/parser.OpenStream(0xc000961dd0?, 0xc0039704c0?, 0xc0025840c0?, 0xca6e?)
        /go/pkg/mod/www.velocidex.com/golang/go-ntfs@v0.1.2-0.20230508135129-d367f954db15/parser/easy.go:356 +0x207
www.velocidex.com/golang/velociraptor/accessors/ntfs.(*MFTFileSystemAccessor).OpenWithOSPath(0xc0003c5230, 0xc0021e8d20?)
        /velociraptor-build/velociraptor/accessors/ntfs/mft.go:139 +0x127
www.velocidex.com/golang/velociraptor/vql/networking.(*UploadFunction).Call(0xc0030efb30?, {0x2d632b0, 0xc00329be40}, {0x2d7f028, 0xc0021e8d20}, 0x23c7ba0?)
        /velociraptor-build/velociraptor/vql/networking/upload.go:85 +0x36f
www.velocidex.com/golang/vfilter.(*_SymbolRef).callFunction(0xc001330de0, {0x2d632b0?, 0xc00329be40}, {0x2d7f028?, 0xc0021e8d20}, {0x2d58f48?, 0x3c17710})
        /go/pkg/mod/www.velocidex.com/golang/vfilter@v0.0.0-20230425012353-b6d0a982ca34/vfilter.go:1730 +0x583
www.velocidex.com/golang/vfilter.(*_SymbolRef).Reduce(0xc001330de0, {0x2d632b0, 0xc00329be40}, {0x2d7f028, 0xc0021e8d20})
        /go/pkg/mod/www.velocidex.com/golang/vfilter@v0.0.0-20230425012353-b6d0a982ca34/vfilter.go:1584 +0x1c6
www.velocidex.com/golang/vfilter.(*_Value).Reduce(0xc00041a400, {0x2d632b0, 0xc00329be40}, {0x2d7f028, 0xc0021e8d20})
        /go/pkg/mod/www.velocidex.com/golang/vfilter@v0.0.0-20230425012353-b6d0a982ca34/vfilter.go:1472 +0x14f
www.velocidex.com/golang/vfilter.(*_MemberExpression).Reduce(0xc000249a00, {0x2d632b0, 0xc00329be40}, {0x2d7f028?, 0xc0021e8d20?})
        /go/pkg/mod/www.velocidex.com/golang/vfilter@v0.0.0-20230425012353-b6d0a982ca34/vfilter.go:1159 +0x56
www.velocidex.com/golang/vfilter.(*_MultiplicationExpression).Reduce(0xc000249a80, {0x2d632b0, 0xc00329be40}, {0x2d7f028?, 0xc0021e8d20?})
        /go/pkg/mod/www.velocidex.com/golang/vfilter@v0.0.0-20230425012353-b6d0a982ca34/vfilter.go:1408 +0x53
www.velocidex.com/golang/vfilter.(*_AdditionExpression).Reduce(0xc000249c00, {0x2d632b0, 0xc00329be40}, {0x2d7f028?, 0xc0021e8d20?})
        /go/pkg/mod/www.velocidex.com/golang/vfilter@v0.0.0-20230425012353-b6d0a982ca34/vfilter.go:1323 +0x53
www.velocidex.com/golang/vfilter.(*_ConditionOperand).Reduce(0xc001e10c60, {0x2d632b0, 0xc00329be40}, {0x2d7f028?, 0xc0021e8d20?})
        /go/pkg/mod/www.velocidex.com/golang/vfilter@v0.0.0-20230425012353-b6d0a982ca34/vfilter.go:1361 +0x85
www.velocidex.com/golang/vfilter.(*_OrExpression).Reduce(0xc000249c40, {0x2d632b0, 0xc00329be40}, {0x2d7f028?, 0xc0021e8d20?})
        /go/pkg/mod/www.velocidex.com/golang/vfilter@v0.0.0-20230425012353-b6d0a982ca34/vfilter.go:1285 +0x56
www.velocidex.com/golang/vfilter.(*_AndExpression).Reduce(0xc000249c80, {0x2d632b0, 0xc00329be40}, {0x2d7f028?, 0xc0021e8d20?})
        /go/pkg/mod/www.velocidex.com/golang/vfilter@v0.0.0-20230425012353-b6d0a982ca34/vfilter.go:1253 +0x4a
www.velocidex.com/golang/vfilter.(*_AliasedExpression).Reduce(0xc006529d20?, {0x2d632b0?, 0xc00329be40?}, {0x2d7f028?, 0xc0021e8d20?})
        /go/pkg/mod/www.velocidex.com/golang/vfilter@v0.0.0-20230425012353-b6d0a982ca34/vfilter.go:699 +0x9f
www.velocidex.com/golang/vfilter.(*_SelectExpression).Transform.func2({0x2d632b0, 0xc00329be40}, {0xc0030efb40?, 0x6?})
        /go/pkg/mod/www.velocidex.com/golang/vfilter@v0.0.0-20230425012353-b6d0a982ca34/vfilter.go:955 +0x5b
www.velocidex.com/golang/vfilter.MaterializedLazyRow({0x2d632b0, 0xc00329be40}, {0x255bc20?, 0xc0000cb030?}, {0x2d7f028, 0xc0021e8be0})
        /go/pkg/mod/www.velocidex.com/golang/vfilter@v0.0.0-20230425012353-b6d0a982ca34/lazy.go:110 +0x1d0
www.velocidex.com/golang/vfilter.(*_Select).processSingleRow(0xc001331f20, {0x2d632b0, 0xc00329be40}, {0x2d7f028, 0xc001368960}, {0x27a8080, 0xc001368960}, 0xc00399efc0)
        /go/pkg/mod/www.velocidex.com/golang/vfilter@v0.0.0-20230425012353-b6d0a982ca34/vfilter.go:581 +0x171
www.velocidex.com/golang/vfilter.(*_Select).Eval.func3()
        /go/pkg/mod/www.velocidex.com/golang/vfilter@v0.0.0-20230425012353-b6d0a982ca34/vfilter.go:563 +0x179
created by www.velocidex.com/golang/vfilter.(*_Select).Eval
        /go/pkg/mod/www.velocidex.com/golang/vfilter@v0.0.0-20230425012353-b6d0a982ca34/vfilter.go:546 +0x30a```

**Comment by clayscode** [comment]
Hmm, 0.7.0 doesn't seem to crash. And also we can only reproduce this bug on Windows 11, not other operating systems.

**Comment by scudette** [comment]
This should be fixed in 0.7.0-3
